pkg/config/config.go

```diff
@@ -105,17 +105,61 @@ func New(opts ...Option) *config {
 	return c
 }
 
+// Load merges a configuration file with the current configuration, ensuring
+// that values set via the config flag (-c or --config) take precedence.
+// This allows users to override config values in multiple ways:
+// - Provide one or more config files
+// - Specify individual config values (e.g., "-c WebUI.Enabled=true")
+// - Use a combination of files and individual values
+// The function carefully merges these sources, maintaining the correct
+// precedence: flag values > file config in bacalhau repo > default values.
 func (c *config) Load(path string) error {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		// if the config file doesn't exist then we obviously cannot load it
-		return fmt.Errorf("config file not found at at path: %q: %w", path, err)
+		return fmt.Errorf("config file not found at path: %q: %w", path, err)
```

Comment:
Actually, after taking another look at how we initialize configs, I am not sure if `Load` is needed anymore. Previously we weren't allowing the users to define a config file and we were calling `Load` when we open the repo to read the configs and populate viper. But now you are parsing `--config/-c` flags and loading the global viper instance with the configs. 

I don't think we should expect a `BACALHAU_DIR/config.yaml` default config if non is provided, but maybe we can do that for backward compatibility if we need to. Still, even if we do that, it should just mean a default `--config` value and load its content the same as if a user provided `--config` file

Some of my comments below about Load implementation might be outdated if we decide to remove it

Reply:
In our sync/conversation yesterday we decided we'll continue to support a default config file location. For now this will be `BACALHAU_DIR`, but in the future will change to `$XDG_CONFIG_HOME` (e.g. `~/.config/bacalhau/`).

For posterity: 
1. It was stated there is value in supporting a default config file path, allowing users to simply edit a file - in a canonical location, then run bacalhau w/o needing to provide a `--config` flag each time. 
2. Additionally, commands like `bacalhau serve` will _require_ some config to operate since fields like `orchestrators` are required for a node to operate.

With that in mind, _if_ a config file is present in `BACALHAU_DIR` then we may treat it as a base config. Values passed via a `--config` flag must take precedence over values present in the default config. This continues to be true for other flag and environment values as well.

Reply:
I think the difference is you are treating the default config path as a base config that will always be loaded even if the user provided their own config. I am looking at it as just a fallback location to load the config _iff_ no file was provided by the user. It might be a bit confusing where we resolve the configs as `user_config_file -> default_config_path -> hardcoded_default_values`

I keep referring to it as a default config path and not default config file if it helps with avoiding confusion. If the user provided their own using `-c/--config` flag, then we don't attempt to look for a config under the default config path

Reply:
I have modified the loading of config files from various locations as you've suggested. Details are covered in the PR description under `Handling of Configuration Files in Bacalhau`.